Sources/Sentry/Profiling/SentryLegacyProfiler.mm

```diff
@@ -0,0 +1,92 @@
+#import "SentryLegacyProfiler.h"
+
+#if SENTRY_TARGET_PROFILING_SUPPORTED
+
+#    import "SentryLog.h"
+#    import "SentryMetricProfiler.h"
+#    import "SentryProfiledTracerConcurrency.h"
+#    import "SentryProfiler+Private.h"
+#    include <mutex>
+
+#    pragma mark - Private
+
+namespace {
+/** @warning: Must be used from a synchronized context. */
+std::mutex _unsafe_gLegacyProfilerLock;
+
+/** @warning: Must be used from a synchronized context. */
+SentryProfiler *_Nullable _unsafe_gLegacyProfiler;
+
+/** @warning: Must be called from a synchronized context. */
+BOOL
+_unsafe_isRunning(void)
```

Comment:
`l`: Maybe you could rename this and other unsafe prefixes to `non_synched_isRunning` or `call_from_synched` or something similar because unsafe doesn't reflect that you have to call it from a synchronized context.

Reply:
Yeah that's fair!

Reply:
I went with a prefix of `_threadUnsafe_`